app/pages/Account/functions.js

```diff
@@ -0,0 +1,313 @@
+    return true;
+  }
+
+  // If nobody revealed their bids, start over.
+  // if (auction.owner.isNull())
+  if (auction.owner.hash === ZERO_HASH && auction.owner.index === 0xffffffff) {
+    return true;
+  }
+
+  return false;
+}
+
+async function getStatsFromBids() {
+  // Live Auctions
+  let biddingHNS = 0;
+  let biddingNum = 0;
+
+  // In Reveals
+  let revealingHNS = 0;
+  let revealingNum = 0;
+
+  // Actionable
+  let revealableHNS = 0;
+  let revealableNum = 0;
+  let revealableBlock = null;
+  let redeemableHNS = 0;
+  let redeemableNum = 0;
+  let registerableHNS = 0;
+  let registerableNum = 0;
+
+  // All bids
+  const bids = await walletClient.getBids();
+  console.log("bids", bids);
+
+  for (let bid of bids) {
+    // Don't bother if not own bid
+    if (!bid.own) {
+      continue;
+    }
+
+    const blind = bid.lockup - bid.value;
+    const auction = await walletClient.getAuctionInfo(bid.name);
+    console.log(auction);
+
+    // Full bid + blind is locked up in the bidding phase
+    if (auction.state === "BIDDING") {
+      biddingNum++;
+      biddingHNS += bid.lockup;
+      continue;
+    }
+
+    // Only care about auctions in REVEAL and CLOSED
+    if (auction.state !== "REVEAL" && auction.state !== "CLOSED") {
+      continue;
+    }
+
+    // If in reveal, first assume that the bid is not yet revealed and the whole amount is locked up
+    // Subtract later if a reveal is found
+    if (auction.state === "REVEAL") {
+      revealingHNS += bid.lockup;
+      revealingNum++;
+    }
+
+    // Search for reveal of this bid
+    let isRevealed = false;
+    let revealPrevOut = null;
+    for (let reveal of auction.reveals) {
+      const revealTx = await walletClient.getTX(reveal.prevout.hash);
+      console.log("revealTx", auction.name, bid.value, revealTx);
+      const inputIdx = revealTx.tx.inputs.reduce((cur, input, idx) => {
+        if (cur != null) return cur;
+        if (
+          input.prevout.hash === bid.prevout.hash &&
+          input.prevout.index === bid.prevout.index
+        )
+          return idx;
+        return null;
+      }, null);
+      if (inputIdx != null) {
+        // This reveal is for the current bid => Bid is revealed
+        isRevealed = true;
+        revealPrevOut = { hash: reveal.prevout.hash, index: inputIdx };
+        break;
+      }
+    }
+
+    // If bid not yet revealed and is in reveal period
+    if (!isRevealed) {
+      if (auction.state === "REVEAL") {
+        revealableHNS += blind;
+        revealableNum++;
+        if (
+          revealableBlock === null ||
+          auction.stats.blocksUntilClose < revealableBlock
+        ) {
+          revealableBlock = auction.stats.blocksUntilClose;
+        }
+      }
+      continue;
+    }
+
+    // At this point, the bid is revealed (at least)
+
+    // If still in reveal period, the blind is no longer locked
+    if (auction.state === "REVEAL") {
+      revealingHNS -= blind;
```

Comment:
![Screen Shot 2021-06-11 at 9 23 01 PM](https://user-images.githubusercontent.com/8507735/121695554-0137f080-ca80-11eb-8eca-93cba508fdb2.png)

I found an edge case where there are `revealingHNS` but no `revealableHNS`.  I think you can likely combine the two variables here.